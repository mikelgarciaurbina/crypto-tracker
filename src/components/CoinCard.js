import React from 'react';
import { Image, StyleSheet, Text, View } from 'react-native';

import { images } from '../Utils/CoinIcons';

const styles = StyleSheet.create({
  bold: {
    fontWeight: 'bold',
  },
  container: {
    display: 'flex',
  },
  image: {
    height: 40,
    width: 40,
  },
});

const { container, image, bold } = styles;

const CoinCard = ({ coin_name, percent_change_24h, percent_change_7d, price_usd, symbol }) => {
  return (
    <View style={container}>
      <Image
        style={image}
        source={{ uri: images[symbol] }}
      />
      <Text>{symbol}</Text>
      <Text>{coin_name}</Text>
      <Text>{price_usd} <Text style={bold}>$</Text></Text>
      <Text>Change past 24 hours: {percent_change_24h}</Text>
      <Text>Change past 7 days: {percent_change_7d}</Text>
    </View>
  );
};

export default CoinCard;
