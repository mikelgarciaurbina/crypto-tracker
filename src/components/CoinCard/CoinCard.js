import React from 'react';
import PropTypes from 'prop-types';
import { View, Text, Image } from 'react-native';

import { C } from '../../constants';
import styles from './CoinCard.styles';

const { images } = C;
const {
  container,
  image,
  moneySymbol,
  upperRow,
  coinSymbol,
  coinNameText,
  coinPrice,
  statisticsContainer,
  seperator,
  percentChangePlus,
  percentChangeMinus,
} = styles;

const CoinCard = ({
  symbol,
  coinName,
  priceUsd,
  percentChange24h,
  percentChange7d,
}) => (
  <View style={container}>
    <View style={upperRow}>
      <Image style={image} source={images[symbol]} />
      <Text style={coinSymbol}>{symbol}</Text>
      <Text style={seperator}>|</Text>
      <Text style={coinNameText}>{coinName}</Text>
      <Text style={coinPrice}>
        {priceUsd}
        <Text style={moneySymbol}> $ </Text>
      </Text>
    </View>
    <View style={statisticsContainer}>
      <Text>
        {'24h: '}
        <Text
          style={percentChange24h < 0 ? percentChangeMinus : percentChangePlus}
        >
          {percentChange24h} %
        </Text>
      </Text>
      <Text>
        {'7d: '}
        <Text
          style={percentChange7d < 0 ? percentChangeMinus : percentChangePlus}
        >
          {percentChange7d} %
        </Text>
      </Text>
    </View>
  </View>
);
CoinCard.propTypes = {
  symbol: PropTypes.string,
  coinName: PropTypes.string,
  priceUsd: PropTypes.string,
  percentChange24h: PropTypes.string,
  percentChange7d: PropTypes.string,
};
CoinCard.defaultProps = {
  symbol: 'BTC',
  coinName: 'Bitcoin',
  priceUsd: '8000',
  percentChange24h: '5',
  percentChange7d: '1',
};

export default CoinCard;
