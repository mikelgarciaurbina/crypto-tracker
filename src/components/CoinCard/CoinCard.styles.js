import { StyleSheet } from 'react-native';

import { THEME } from '../../constants';

export default StyleSheet.create({
  container: {
    display: 'flex',
    marginBottom: 20,
    borderBottomColor: THEME.GREY,
    borderBottomWidth: 3,
    padding: 20,
  },
  upperRow: {
    display: 'flex',
    flexDirection: 'row',
    marginBottom: 15,
  },
  coinSymbol: {
    marginTop: 10,
    marginLeft: 20,
    marginRight: 5,
    fontWeight: 'bold',
  },
  coinName: {
    marginTop: 10,
    marginLeft: 5,
    marginRight: 20,
  },
  seperator: {
    marginTop: 10,
  },
  coinPrice: {
    marginTop: 10,
    marginLeft: 'auto',
    marginRight: 10,
    fontWeight: 'bold',
  },
  image: {
    width: 35,
    height: 35,
  },
  moneySymbol: {
    fontWeight: 'bold',
  },
  statisticsContainer: {
    display: 'flex',
    borderTopColor: THEME.LIGHT_GREY,
    borderTopWidth: 2,
    padding: 10,
    flexDirection: 'row',
    justifyContent: 'space-around',
  },
  percentChangePlus: {
    color: THEME.GREEN,
    fontWeight: 'bold',
    marginLeft: 5,
  },
  percentChangeMinus: {
    color: THEME.RED,
    fontWeight: 'bold',
    marginLeft: 5,
  },
});
