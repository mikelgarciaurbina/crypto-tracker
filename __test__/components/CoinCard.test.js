import React from 'react';
import renderer from 'react-test-renderer';

import CoinCard from '../../src/components/CoinCard';

test('renders correctly', () => {
  const tree = renderer.create(<CoinCard />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly with positive values', () => {
  const tree = renderer.create(<CoinCard symbol="BTC" coinName="Bitcoin" priceUsd="8000" percentChange24h="5" percentChange7d="15" />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly with negative values', () => {
  const tree = renderer.create(<CoinCard symbol="BTC" coinName="Bitcoin" priceUsd="8000" percentChange24h="-5" percentChange7d="-15" />).toJSON();
  expect(tree).toMatchSnapshot();
});
