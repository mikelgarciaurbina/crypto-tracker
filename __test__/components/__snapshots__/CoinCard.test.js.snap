// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "#e5e5e5",
      "borderBottomWidth": 3,
      "display": "flex",
      "marginBottom": 20,
      "padding": 20,
    }
  }
>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "marginBottom": 15,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 35,
          "width": 35,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "fontWeight": "bold",
          "marginLeft": 20,
          "marginRight": 5,
          "marginTop": 10,
        }
      }
    >
      BTC
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "marginTop": 10,
        }
      }
    >
      |
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "marginLeft": 5,
          "marginRight": 20,
          "marginTop": 10,
        }
      }
    >
      Bitcoin
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "fontWeight": "bold",
          "marginLeft": "auto",
          "marginRight": 10,
          "marginTop": 10,
        }
      }
    >
      8000
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
         $ 
      </Text>
    </Text>
  </View>
  <View
    style={
      Object {
        "borderTopColor": "#FAFAFA",
        "borderTopWidth": 2,
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-around",
        "padding": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
    >
      24h: 
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#00BFA5",
            "fontWeight": "bold",
            "marginLeft": 5,
          }
        }
      >
        5
         %
      </Text>
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
    >
      7d: 
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#00BFA5",
            "fontWeight": "bold",
            "marginLeft": 5,
          }
        }
      >
        1
         %
      </Text>
    </Text>
  </View>
</View>
`;

exports[`renders correctly with negative values 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "#e5e5e5",
      "borderBottomWidth": 3,
      "display": "flex",
      "marginBottom": 20,
      "padding": 20,
    }
  }
>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "marginBottom": 15,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 35,
          "width": 35,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "fontWeight": "bold",
          "marginLeft": 20,
          "marginRight": 5,
          "marginTop": 10,
        }
      }
    >
      BTC
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "marginTop": 10,
        }
      }
    >
      |
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "marginLeft": 5,
          "marginRight": 20,
          "marginTop": 10,
        }
      }
    >
      Bitcoin
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "fontWeight": "bold",
          "marginLeft": "auto",
          "marginRight": 10,
          "marginTop": 10,
        }
      }
    >
      8000
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
         $ 
      </Text>
    </Text>
  </View>
  <View
    style={
      Object {
        "borderTopColor": "#FAFAFA",
        "borderTopWidth": 2,
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-around",
        "padding": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
    >
      24h: 
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#DD2C00",
            "fontWeight": "bold",
            "marginLeft": 5,
          }
        }
      >
        -5
         %
      </Text>
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
    >
      7d: 
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#DD2C00",
            "fontWeight": "bold",
            "marginLeft": 5,
          }
        }
      >
        -15
         %
      </Text>
    </Text>
  </View>
</View>
`;

exports[`renders correctly with positive values 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "#e5e5e5",
      "borderBottomWidth": 3,
      "display": "flex",
      "marginBottom": 20,
      "padding": 20,
    }
  }
>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "marginBottom": 15,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 35,
          "width": 35,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "fontWeight": "bold",
          "marginLeft": 20,
          "marginRight": 5,
          "marginTop": 10,
        }
      }
    >
      BTC
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "marginTop": 10,
        }
      }
    >
      |
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "marginLeft": 5,
          "marginRight": 20,
          "marginTop": 10,
        }
      }
    >
      Bitcoin
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "fontWeight": "bold",
          "marginLeft": "auto",
          "marginRight": 10,
          "marginTop": 10,
        }
      }
    >
      8000
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
         $ 
      </Text>
    </Text>
  </View>
  <View
    style={
      Object {
        "borderTopColor": "#FAFAFA",
        "borderTopWidth": 2,
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-around",
        "padding": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
    >
      24h: 
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#00BFA5",
            "fontWeight": "bold",
            "marginLeft": 5,
          }
        }
      >
        5
         %
      </Text>
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
    >
      7d: 
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#00BFA5",
            "fontWeight": "bold",
            "marginLeft": 5,
          }
        }
      >
        15
         %
      </Text>
    </Text>
  </View>
</View>
`;
